independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/MavenRepoInitializer.java

```diff
@@ -197,18 +221,49 @@ public static DefaultRepositorySystemSession newSession(RepositorySystem system,
 
     public static List<RemoteRepository> getRemoteRepos(Settings settings) throws AppModelResolverException {
         final List<RemoteRepository> remotes = new ArrayList<>();
+
+        List<String> activeProfiles = Collections.emptyList();
+        List<String> inactiveProfiles = Collections.emptyList();
+        if(mvnArgs != null) {
+            final String[] profileOptionValues = mvnArgs.getOptionValues(CLIManager.ACTIVATE_PROFILES);
+            if (profileOptionValues != null && profileOptionValues.length > 0) {
+                activeProfiles = new ArrayList<>();
+                inactiveProfiles = new ArrayList<>();
+                for (String profileOptionValue : profileOptionValues) {
+                    final StringTokenizer profileTokens = new StringTokenizer(profileOptionValue, ",");
+                    while (profileTokens.hasMoreTokens()) {
+                        final String profileAction = profileTokens.nextToken().trim();
+                        if (profileAction.startsWith("-") || profileAction.startsWith("!")) {
+                            inactiveProfiles.add(profileAction.substring(1));
+                        } else if (profileAction.startsWith("+")) {
+                            activeProfiles.add(profileAction.substring(1));
+                        } else {
+                            activeProfiles.add(profileAction);
+                        }
+                    }
+                }
+            }
+        }
+
         for (Profile profile : settings.getProfiles()) {
-            if (profile.getActivation() != null && profile.getActivation().isActiveByDefault()) {
+            if(inactiveProfiles.contains(profile.getId())) {
+                continue;
+            }
+            if (activeProfiles.contains(profile.getId())
+                    || profile.getActivation() != null && profile.getActivation().isActiveByDefault()) {
```

Comment:
any chance you could add parentheses here for the second grouped condition? Not that it's incorrect but it's better to be extra clear about it.